mitiq/benchmarks/random_clifford_t_circuits.py

```diff
@@ -0,0 +1,114 @@
+# Copyright (C) 2020 Unitary Fund
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""Functions for generating randomized benchmarking circuits."""
+from typing import List, Optional, cast
+import copy
+import numpy as np
+
+from cirq.experiments.qubit_characterizations import (
+    _single_qubit_cliffords,
+    _random_single_q_clifford,
+    _random_two_q_clifford,
+    _gate_seq_to_mats,
+    _find_inv_matrix,
+    _two_qubit_clifford_matrices,
+)
+from typing import Sequence
+from cirq import LineQubit, ops,circuits
+from mitiq import QPROGRAM
+from mitiq.interface import convert_from_mitiq
+
+
+def generate_random_clifford_t_circuits(
+    n_qubits: int,
+    num_T: int,
+    num_cliffords: int,
+    trials: int,
+    return_type: Optional[str] = None,
+) -> List[QPROGRAM]:
+    """Returns a list of random clifford + t circuits
+
+    Args:
+        num_T: The number of t gates in the circuit
+        n_qubits: The number of qubits. Can be either 1 or 2.
+        num_cliffords: The number of Clifford group elements in the
+            random circuits. This is proportional to the depth per circuit.
+        trials: The number of random circuits generated. The generation scheme is ran for (trials) times and a circuit generated each time.
+        return_type: String which specifies the type of the
+            returned circuits. See the keys of
+            ``mitiq.SUPPORTED_PROGRAM_TYPES`` for options. If ``None``, the
+            returned circuits have type ``cirq.Circuit``.
+
+    Returns:
+        A list of clifford + t circuits.
+    """
+    if n_qubits not in [1]:
```

Comment:
This seems to work for 1 qubit only. Ideally we want the function to work for arbitrary qubits >= 1.

Reply:
I have updated the main function according to your idea. I have kept the _random_single_q_clifford_T just in case uniform sampling has to be added in a future iteration.